### PR TITLE
Housing counselor: Update JS and cleanup

### PR DESCRIPTION
**Pull request by anselmbradford**
## Removals

- Removes jQuery from housing counselor page.

## Changes

- Removes unnecessary markup from housing counselor page.
- Removes unused styles from housing counselor page.
- Fixes JS for clicking map markers to scroll to result list items.
- Top aligns result list items.
- Moves map marker icons into find a housing counselor app assets.
- Adds more JS test coverage.
- Makes it so hud.js module doesn't auto-initialize the map (auto-initialization is an anti-pattern in my opinion).

## Testing

1. `gulp clean && gulp build`
2. Visit http://localhost:8000/find-a-housing-counselor/ and enter a zip code.
3. Click a map marker and observe it scrolls to the result item.

### CODE REVIEW

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html [hunk 1]
```diff
@@ -180,42 +177,52 @@ <h2 class="h4">
                             </thead>
                             <tbody>
                                 {% for counselor in api_json.counseling_agencies %}
-                                <tr>
-                                    <td data-label="Agency">
+                                <tr id="hud-result-{{ loop.index }}">
+                                    <td>
```
Comment (niqjohnson) [inline cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: I think the `data-label` needs to stay there so the label will appear when the table columns stack on top of each other on small screens

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html [hunk 2]
```diff
@@ -180,42 +177,52 @@ <h2 class="h4">
...
+                                                <b>Phone:</b> {{ counselor.phone1 }}
+                                                <br>
+                                                <b>Email Address:</b> {{ counselor.email }}
+                                                <br>
                                                 <b>Languages:</b> {{ counselor.languages | join( ', ' ) }}
                                             </p>
                                         </div>
                                     </td>
-                                    <td data-label="Services">
+                                    <td>
```
Comment (niqjohnson) [inline cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: I think the `data-label` needs to stay there so the label will appear when the table columns stack on top of each other on small screens

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html [hunk 3]
```diff
@@ -180,42 +177,52 @@ <h2 class="h4">
...
                                                 {% endfor %}
                                             </ul>
-                                            <p></p>
                                         </div>
                                     </td>
-                                    <td data-label="Distance">
-                                        <div class="rich-text">
-                                            <p>{{ counselor.distance | round( 1 ) }} miles</p>
-                                        </div>
+                                    <td>
```
Comment (niqjohnson) [inline cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: I think the `data-label` needs to stay there so the label will appear when the table columns stack on top of each other on small screens

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html (hunk 1 above)
Comment (anselmbradford) [reply cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: Ahh hah, I was wondering about these. Should have checked in CF!

Comment (anselmbradford) [reply cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: Fixed!

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html (hunk 2 above)
Comment (anselmbradford) [reply cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: Fixed!

File: cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html (hunk 3 above)
Comment (anselmbradford) [reply cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html]: Fixed!

**Comment by anselmbradford** [comment]
@niqjohnson Added one slight addition to make it so hud.js doesn't auto-initialize so that the tests can examine it without it running unexpected code. I'll merge this after 🦃 day.

**Pull request merged**